tests/std/tests/VSO_1062649_overaligned_function/test.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <functional>
+
+#ifndef __cpp_aligned_new
+#error This test is only valid when C++17 over-aligned allocation is supported
+#endif
+
+#pragma warning(disable : 4324) // "was padded due to alignment specifier", thanks, Captain
+
+struct alignas(2 * alignof(std::max_align_t)) overaligned_t {
+    char non_empty;
+
+    void operator()(const void* const storage, const std::size_t storage_size) const {
+        const unsigned char* storage_bytes = reinterpret_cast<const unsigned char*>(storage);
+        const unsigned char* this_bytes    = reinterpret_cast<const unsigned char*>(this);
+
+        // platform-specific behavior not covered by Standard C++, but fine for such test

```

Comment:
Hmmm I think this would be closer to spec if they were uintptr_ts to begin with rather than pointers. Plus then you could get rid of the extra cast on line 24.

Reply:
addressed